tests/types.rs

```diff
@@ -129,10 +129,54 @@ fn hex_encoding() {
 
 #[test]
 #[cfg(feature = "serde")]
-fn test_key_with_big_array() {
+fn test_key_with_big_array_serde() {
     let rng = &mut StdRng::seed_from_u64(8586);
-    let s: Bytes64 = rng.gen();
-    let j = serde_json::to_string(&s).unwrap();
-    let s_back = serde_json::from_str(&j).unwrap();
-    assert!(s == s_back);
+    let bytes64: Bytes64 = rng.gen();
+    let bytes64_t = bincode::serialize(&bytes64).expect("Failed to serialize Bytes64");
+    let bytes64_t: Bytes64 =
+        bincode::deserialize(&bytes64_t).expect("Failed to deserialize Bytes64");
+    assert_eq!(bytes64, bytes64_t);
+}
+
+#[test]
+#[cfg(feature = "serde")]
+fn test_key_serde() {
+    let rng = &mut StdRng::seed_from_u64(8586);
+    let adr: Address = rng.gen();
+    let ast_id: AssetId = rng.gen();
+    let contract_id: ContractId = rng.gen();
+    let bytes4: Bytes4 = rng.gen();
+    let bytes8: Bytes8 = rng.gen();
+    let bytes32: Bytes32 = rng.gen();
```

Comment:
`Bytes64` was skipped

Reply:
Since there is a separate `test_key_with_big_array_serde` that tests `Bytes64` I did not include it in this test. Should I remove the separate test (`test_key_with_big_array_serde`) and add it here?

Reply:
I think its better; there is no clear distinction between them in the exposed API - even tho the internal implementation differs